### ISSUE TITLE
Build from AUR?

### ISSUE THREAD

**Issue by mysteryx93** [AkarinVS/vapoursynth-plugin#7]
Is this package available on Linux in the AUR? Every other package was there... only missing this one

**Comment by AkarinVS** [comment]
The included meson.build should work under Linux. Just make sure to have
the correct llvm version (10 to 12, and 13 is not yet supported.)

**Comment by NSQY** [comment]
If you want to compile this under Arch, you will need to downgrade LLVM since they've been shipping 13.x for months now. 
There is a helper tool called downgrade which I have used to compile this plugin: https://aur.archlinux.org/packages/downgrade/

For me, I had to downgrade to `34)  llvm    12.0.1  5` (the 5th version of 12.0.1 in the archives). For whatever reason, this plugin was not compiling with other version of LLVM 12.x. This may have occurred due to a packing error, I do not know.

**Comment by NSQY** [comment]
CC: @sl1pkn07 
You may be interested in this. This plugin is a soft requirement for https://github.com/DJATOM/vapoursynth-preview/blob/api-v4-support/vspreview/main.py and a few other functions.

**Comment by sl1pkn07** [comment]
ok. when go to home

greetings

**Comment by sl1pkn07** [comment]
done

https://aur.archlinux.org/packages/vapoursynth-plugin-vsakarin-git/

**Comment by sl1pkn07** [comment]
pacman not see the aur repository. you need download the PKGBUILD by hand and build yourself with makepkg or use a helper like yay or something

greetings

**Comment by mysteryx93** [comment]
ah but my "Add/Remove Software" supports it. Found the "Update" button. Working!

**Comment by mysteryx93** [comment]
The AUR build doesn't seem to work

> AttributeError: No attribute with the name akarin exists. Did you mistype a plugin namespace?

It's not getting loaded like the other plugins.

**Comment by mysteryx93** [comment]
OK that's strange. Simple script.

    clip = core.akarin.Expr(clip, "x")

> AttributeError: No attribute with the name akarin exists. Did you mistype a plugin namespace?

I do have `/usr/lib/vapoursynth/libakarin.so` that takes 44.9MB, alongside all the other plugins that do get loaded. Something wrong with that one?

**Comment by mysteryx93** [comment]
aahhh I'd bet that the problem with vsakarin is that it uses LLVM to compile expressions on-the-fly, but needs an older version. That means that I need 2 versions of LLVM installed on my system side-by-side? How do I do that on arch-based linux?

**Comment by sl1pkn07** [comment]
that spression is compiled by library or program? with library maybe with `LD_LIBRARY_PATH=>path of the llvm libraries<`.  if use a executable, IDK

**Comment by mysteryx93** [comment]
Probably we need `llvmlibs-11` library. But simply installing that dependency still doesn't work.

**Comment by mysteryx93** [comment]
Actually the problem may be that it was built with dynamic link to LLVM instead of being statically linked.

**Comment by mysteryx93** [comment]
I used downgrade on llvm; llvm-libs does need to be downgraded too. Once manually compiled, then it works. It still works after re-updating llvm and llvm-libs, so it will need to be tested on another computer that didn't have that manual intervention.

**Comment by AkarinVS** [comment]
I have pushed a commit to support LLVM 13, so building on Linux might be
easier as you don't need to install an older version of LLVM.

**Comment by sl1pkn07** [comment]
@mysteryx93 try now with the updated pkgbuild

**Comment by quietvoid** [comment]
For what it's worth I'm getting this when `vapoursynth-editor` tries to load:
```
$ vsedit
: CommandLine Error: Option 'reuse-storage-in-coroutine-frame' registered more than once!
LLVM ERROR: inconsistency in registered CommandLine options
```

Uninstalling `vapoursynth-plugin-vsakarin-git` fixes it.

**Comment by sl1pkn07** [comment]
then is problem of vsedit. try to build the plugin editing the PKGBUILD and add the line `options=('debug')`, build the plugin and install the two packages generated (one with prefix `debug` in the name). and ty again gdb

greetings

**Comment by quietvoid** [comment]
I don't think this is a packaging issue nor a vsedit problem, hence why I'm posting here.
Here's a better backtrace: [vsedit-run.txt](https://github.com/AkarinVS/vapoursynth-plugin/files/8017611/vsedit-run.txt)

**Comment by sl1pkn07** [comment]
![Screenshot_20220207_201826](https://user-images.githubusercontent.com/462213/152856662-833e6bd4-ae9b-4927-8ae3-dc0437358245.png)

**Comment by quietvoid** [comment]
I don't know what else to say, it doesn't work for me.

**Comment by AkarinVS** [comment]
I have seen errors like this in the past, perhaps you have two preloaded
plugins that both use LLVM (perhaps one is statically linked with LLVM, and
another one is dynamically linked with LLVM)
(LLVM registers some global states when the DSO is loaded and the two
copies of LLVM libraries will both be trying to register the same thing and
hence trigger an assertion)

**Comment by quietvoid** [comment]
I've tried removing all plugins except for `libakarin` and it still errors.

**Comment by marillat** [comment]
Under Debian we have the same problem, only LLVM 17 18 and 19 are available.
